### PR TITLE
@W-15316962/W-15316987: [iOS] Update native login sample app to include User Registration/Forgot Password flows

### PR DESCRIPTION
**Pull request by JohnsonEricAtSalesforce**
🤘🏻 _*Ready For Review!_ 🎤

This is a companion to the MSDK implementation in https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/3728

The most interesting part of the code is a handful of utility methods in the layout to help keep the code from being repetitive.

Here are a few useful screenshots.

0  Auth Landing
1  Register
2  Reset Password

### CODE REVIEW

File: iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift [hunk 1]
```diff
@@ -176,15 +196,231 @@ struct NativeLoginView: View {
...
+                        } label: {
+                            Text("Register").frame(minWidth: 150)
+                        }
+                        .buttonStyle(.bordered)
+                        .tint(colorScheme == .dark ? .white : .blue)
+                        .zIndex(2.0)
+                        
+                    case .StartRegistration:
+                        TextField("Email", text: $email)
+                            .autocapitalization(.none)
```
Comment (bbirman) [inline iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift:227]: It looks like these fields have a lot of modifiers in common, could some of them be applied higher up in the hierarchy? or maybe a custom modifier to encapsulate them?

Comment (JohnsonEricAtSalesforce) [reply iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift:227]: I was wanting to add that as well, so I don't need much convincing.  Thanks!  I'll get that up soon.

Comment (JohnsonEricAtSalesforce) [reply iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift]: https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/414/commits/6c7d1c84be7c442457b5518be4914d5e66ed8499
That looks _so_ much *better*!  Thanks, @bbirman!

Comment (JohnsonEricAtSalesforce) [reply iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift]: Maybe there's one thing to mention.  There were some very subtle changes in the top and bottom padding values.  However, I thought it looked better when I made them all consistent across the board.  So, I had padding parameters for the modifiers but then removed them.

Comment (bbirman) [reply iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift]: nice, thanks for trying it out!

**Review by bbirman — APPROVED** [review]

**Pull request merged**